src/control/server/ctl_storage_rpc.go

```diff
@@ -166,128 +170,330 @@ func (c *ControlService) scanScm(ctx context.Context, req *ctlpb.ScanScmReq) (*c
+		if clusterPerTarget < devicesToAdjust[rank].clusterPerTarget {
+			c.log.Debugf("Updating number of clusters per target of rank %d: old=%d new=%d",
+				rank, devicesToAdjust[rank].clusterPerTarget, clusterPerTarget)
+			devicesToAdjust[rank].clusterPerTarget = clusterPerTarget
+		}
+	}
+
+	for _, ctlr := range resp.GetCtrlrs() {
+		engineCfg, err := c.getEngineCfgFromNvmeCtl(ctlr)
+		if err != nil {
+			c.log.Noticef("Skipping NVME controller %s: %s",
+				ctlr.GetPciAddr(), err.Error())
+			continue
+		}
+
+		for idx, dev := range ctlr.GetSmdDevices() {
+			rank := dev.GetRank()
+
+			if dev.GetDevState() != ctlpb.NvmeDevState_NORMAL {
+				c.log.Debugf("SMD device %s (rank %d, ctlr %s) not usable: device state %q",
+					dev.GetUuid(), rank, ctlr.GetPciAddr(), ctlpb.NvmeDevState_name[int32(dev.DevState)])
+				dev.TotalBytes = 0
+				dev.AvailBytes = 0
+				dev.UsableBytes = 0
+				continue
+			}
+
+			if dev.GetClusterSize() == 0 || len(dev.GetTgtIds()) == 0 {
+				c.log.Noticef("SMD device %s (rank %d,  ctlr %s) not usable: missing storage info",
+					dev.GetUuid(), rank, ctlr.GetPciAddr())
+				dev.AvailBytes = 0
+				dev.UsableBytes = 0
+				continue
+			}
+
+			if dev.GetRoleBits() != 0 && (dev.GetRoleBits()&storage.BdevRoleData) == 0 {
+				c.log.Debugf("SMD device %s (rank %d, ctlr %s) not used to store data (Role bits 0x%X)",
+					dev.GetUuid(), rank, ctlr.GetPciAddr(), dev.GetRoleBits())
+				dev.AvailBytes = 0
+				dev.UsableBytes = 0
+				continue
+			}
+
+			c.log.Debugf("Initial available size of SMD device %s (rank %d, ctlr %s): %s (%d bytes)",
+				dev.GetUuid(), rank, ctlr.GetPciAddr(), humanize.Bytes(dev.GetAvailBytes()), dev.GetAvailBytes())
+
+			clusterSize := uint64(dev.GetClusterSize())
+			availBytes := (dev.GetAvailBytes() / clusterSize) * clusterSize
+			if dev.GetAvailBytes() != availBytes {
+				c.log.Debugf("Adjusting available size of SMD device %s (rank %d, ctlr %s): from %s (%d Bytes) to %s (%d bytes)",
+					dev.GetUuid(), rank, ctlr.GetPciAddr(),
+					humanize.Bytes(dev.GetAvailBytes()), dev.GetAvailBytes(),
+					humanize.Bytes(availBytes), availBytes)
+				dev.AvailBytes = availBytes
+			}
+
+			dataClusterCount := dev.GetAvailBytes() / clusterSize
+			if dev.GetRoleBits() == 0 {
+				c.log.Debugf("No meta-data stored on SMD device %s (rank %d, ctlr %s)",
+					dev.GetUuid(), rank, ctlr.GetPciAddr())
+				addDeviceToAdjust(ctlr, idx, rank, dataClusterCount)
+				continue
+			}
+
+			getClusterCount := func(sizeBytes uint64, targetNb uint64) uint64 {
+				clusterCount := sizeBytes / clusterSize
+				if sizeBytes%clusterSize != 0 {
+					clusterCount += 1
+				}
+				return clusterCount * targetNb
+			}
+
+			engineTargetNb := uint64(engineCfg.TargetCount)
+			if (dev.GetRoleBits() & storage.BdevRoleMeta) != 0 {
+				metadataClusterCount := getClusterCount(dev.GetMetaSize(), engineTargetNb)
+				c.log.Debugf("Removing %d Metadata clusters from the usable size of the SMD device %s (rank %d, ctlr %s): ",
```

Comment:
Is all of the debug necessary, it seems like this could be quite noisy.

Reply:
From my side, these debug lines where very useful to understand what is happening and could help the support to understand how the final size of a pool was calculated.
Moreover, they will not be so much printed as they need to be proactively called by the end-user for storage scan or pool creation.
Could it be acceptable to change it to trace ?

Reply:
yes I think that's a good idea

Reply:
- [ ] Reduce debug verbosity